### PR TITLE
visor_operation and visor_result working

### PR DESCRIPTION
**Pull request by regmarcos**
Hi! the visor is working, as you can see in the screenshot. 
The design of the visor will be do in the next PR.
Thanks for your time!

Anotación 2019-11-06 142648

### CODE REVIEW

File: app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java [hunk 1]
```diff
@@ -2,7 +2,54 @@
...
-    private int operando_1;
-    private String operador;
-    private float resultado;
+    private String operand_1;
+    private String operand_2;
+    private String operator;
+    private float result;
+
+    public void cleanVisor() {
+        operand_1 = "";
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java]: Please, remove the hardcode into a StringUtil.

File: app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java [hunk 2]
```diff
@@ -2,7 +2,54 @@
...
+        operand_2 = "";
+        operator = "";
+        result = 0;
+    }
+
+    public CalculatorModel(){
+        operand_1 = "";
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java]: same

File: app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java [hunk 3]
```diff
@@ -3,26 +3,55 @@
...
+        else {
+            model.setOperand_2(model.getOperand_2() + number);
+            view.setVisor(model.getOperand_1() + model.getOperator() + model.getOperand_2());
+        }
     }
 
-    public void onResetButtonPressed(){
+    public void onOperatorPressed(String operator){
 
+        if (model.getOperand_1() != "" && model.getOperand_2()==""){
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java]: Please, remove this hardcode into a SringUtil.

File: app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java [hunk 4]
```diff
@@ -3,26 +3,55 @@
...
+            model.setOperator(operator);
+            view.setVisor(model.getOperand_1()+model.getOperator());
+        }
+        else {
+            Toast.makeText(view , "Please insert the first operand", Toast.LENGTH_SHORT).show();
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java]: The same for this message toast, you will be add in a resources

File: app/src/main/java/com/example/calculadoramvp/mvp/view/CalculatorView.java [hunk 5]
```diff
@@ -1,5 +1,36 @@
...
+public class CalculatorView extends ActivityView {
+
+    @BindView(R.id.visor_operation) TextView visorOperation;
+    @BindView(R.id.visor_result) TextView visorResult;
+
+    public CalculatorView(Activity activity){
+        super(activity);
+        ButterKnife.bind(this,activity);
+
+        //TODO
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/calculadoramvp/mvp/view/CalculatorView.java]: Please add some message into the TODO comment to understand the behavior.

File: app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java (hunk 1 above)
Comment (regmarcos) [reply app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java]: done

File: app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java (hunk 2 above)
Comment (regmarcos) [reply app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java]: donde

File: app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java (hunk 3 above)
Comment (regmarcos) [reply app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java]: done

File: app/src/main/java/com/example/calculadoramvp/mvp/view/CalculatorView.java (hunk 5 above)
Comment (regmarcos) [reply app/src/main/java/com/example/calculadoramvp/mvp/view/CalculatorView.java]: done

File: app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java [hunk 6]
```diff
@@ -3,26 +3,56 @@
...
-    public void onButtonPressed(){
-        //TODO
-        /*IF IS =
-        IN MODEL I REALIZE A COUNT
-        MODIFIED THE VIEW TO SHOW THE RESULT
-        IF IS NOT =
-        IN THE VIEW SHOW NUMBERS*/
+    public void onNumberPressed(String number){
+
+        if (model.getOperand_1()==""){
```
Comment (lucianoeli) [inline app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java]: could use EMPTY_STRING

File: app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java [hunk 7]
```diff
@@ -3,26 +3,56 @@
...
+            model.setOperand_1(number);
+            view.setVisor(number);
+        }
+        else if (model.getOperand_1() != EMPTY_STRING && model.getOperator()=="") {
```
Comment (lucianoeli) [inline app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java]: same

File: app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java [hunk 8]
```diff
@@ -3,26 +3,56 @@
...
+            view.setVisor(model.getOperand_1()+model.getOperator());
+        }
+        else {
+            Toast.makeText(view , TOAST_MSG_FIRST_OPERAND, Toast.LENGTH_SHORT).show();
+        }
     }
 
+    public void onEqualsPressed(){
+
+        if (model.getOperator()=="+"){
```
Comment (lucianoeli) [inline app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java]: could add "+" in stringUtils

File: app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java [hunk 9]
```diff
@@ -1,8 +1,57 @@
 package com.example.calculadoramvp.mvp.model;
 
+import static com.example.calculadoramvp.Utils.EMPTY_STRING;
+
 public class CalculatorModel {
 
-    private int operando_1;
-    private String operador;
-    private float resultado;
+    private String operand_1;
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java]: change the name plase. use the standars without "_"
In this case you can use "firstOperand"

File: app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java [hunk 10]
```diff
@@ -1,8 +1,57 @@
...
+    private String operand_2;
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java]: same

File: app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java [hunk 11]
```diff
@@ -1,8 +1,57 @@
...
+    }
+
+    public CalculatorModel(){
+        operand_1 = EMPTY_STRING;
+        operand_2 = EMPTY_STRING;
+        operator = EMPTY_STRING;
+        result = 0;
+    }
+
+    public void setOperand_1(String operand_1) {
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java]: same

File: app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java [hunk 12]
```diff
@@ -1,8 +1,57 @@
...
+        this.operand_1 = operand_1;
+    }
+
+    public void setOperand_2(String operand_2) {
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java]: same

File: app/src/main/java/com/example/calculadoramvp/mvp/view/CalculatorView.java [hunk 13]
```diff
@@ -1,5 +1,34 @@
...
+import android.app.Activity;
+import android.widget.TextView;
+import com.example.calculadoramvp.R;
 
-}
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+
+
```
Comment (lucianoeli) [inline app/src/main/java/com/example/calculadoramvp/mvp/view/CalculatorView.java]: delete unnecessary empty lines

File: app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java [hunk 14]
```diff
@@ -1,8 +1,57 @@
...
+
+    public void setOperator(String operator) {
+        this.operator = operator;
+    }
+
+    public void setResult(float result) {
+        this.result = result;
+    }
+
+    public String getOperand_1() {
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java]: same

File: app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java [hunk 15]
```diff
@@ -1,8 +1,57 @@
...
+        return operand_1;
+    }
+
+    public String getOperand_2() {
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java]: same

File: app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java (hunk 6 above)
Comment (SartiMau) [inline app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java]: no hardcode pls.. use EMPTY_STRING

File: app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java (hunk 7 above)
Comment (SartiMau) [inline app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java]: same

File: app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java (hunk 8 above)
Comment (SartiMau) [inline app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java]: no hardcode pls.. create a constant PLUS in the StringUtils class

File: app/src/main/java/com/example/calculadoramvp/mvp/view/ActivityView.java [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+        return activityRef.get();
+    }
+
+    @Nullable
+    public Context getContext() {
+        return getActivity();
+    }
+
+    @Nullable
+    public FragmentManager getFragmentManager() {
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculadoramvp/mvp/view/ActivityView.java]: You don't need a FragmentManager here. Please delete it

File: app/src/main/java/com/example/calculadoramvp/mvp/view/CalculatorView.java [hunk 17]
```diff
@@ -1,5 +1,34 @@
...
 
-public class CalculatorView {
+import android.app.Activity;
+import android.widget.TextView;
+import com.example.calculadoramvp.R;
 
-}
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculadoramvp/mvp/view/CalculatorView.java:9]: please delete the unnecessary spaces

File: app/src/main/java/com/example/calculadoramvp/mvp/view/CalculatorView.java [hunk 18]
```diff
@@ -1,5 +1,34 @@
...
+    @BindView(R.id.visor_operation) TextView visorOperation;
+    @BindView(R.id.visor_result) TextView visorResult;
+
+    public CalculatorView(Activity activity){
+        super(activity);
+        ButterKnife.bind(this,activity);
+    }
+
+    public void setVisor(String visor) {
+
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculadoramvp/mvp/view/CalculatorView.java]: same

File: app/src/main/java/com/example/calculadoramvp/mvp/view/CalculatorView.java [hunk 19]
```diff
@@ -1,5 +1,34 @@
...
+        visorOperation.setText(visor);
+    }
+
+
+    public void showResult(float result) {
+
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculadoramvp/mvp/view/CalculatorView.java]: same

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java (hunk 6 above)
Comment (regmarcos) [reply app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java]: done

File: app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java (hunk 7 above)
Comment (regmarcos) [reply app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java]: done

File: app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java (hunk 8 above)
Comment (regmarcos) [reply app/src/main/java/com/example/calculadoramvp/mvp/presenter/CalculatorPresenter.java]: done

File: app/src/main/java/com/example/calculadoramvp/mvp/view/CalculatorView.java (hunk 13 above)
Comment (regmarcos) [reply app/src/main/java/com/example/calculadoramvp/mvp/view/CalculatorView.java]: done

File: app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java (hunk 9 above)
Comment (regmarcos) [reply app/src/main/java/com/example/calculadoramvp/mvp/model/CalculatorModel.java]: done

**Review by lucianoeli — APPROVED** [review]

File: app/src/main/java/com/example/calculadoramvp/mvp/view/CalculatorView.java [hunk 20]
```diff
@@ -1,5 +1,28 @@
...
+    }
+
+    public void setVisor(String visor) {
+        visorOperation.setText(visor);
+    }
+
+    public void showResult(float result) {
+        visorResult.setText(String.valueOf(result));
+    }
+
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculadoramvp/mvp/view/CalculatorView.java:26]: remove unnecessary space

File: app/src/main/res/values/strings.xml [hunk 21]
```diff
@@ -16,4 +16,5 @@
     <string name="sum">+</string>
     <string name="multiply">*</string>
     <string name="share">/</string>
+    <string name="texto">Hello World!</string>
```
Comment (SartiMau) [inline app/src/main/res/values/strings.xml]: is this in use? if not, delete pls

**Review by SartiMau — CHANGES_REQUESTED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Pull request merged**